pkg/daemons/agent/agent_windows.go

```diff
@@ -122,5 +133,77 @@ func kubeletArgs(cfg *config.Agent) map[string]string {
 	if cfg.ProtectKernelDefaults {
 		argsMap["protect-kernel-defaults"] = "true"
 	}
+
 	return argsMap
 }
+
+func waitForSourceVip(networkName string) string {
+	for range time.Tick(time.Second * 5) {
+		network, err := hcsshim.GetHNSNetworkByName(networkName)
+		if err != nil {
+			logrus.WithError(err).Warning("can't find HNS network, retrying", networkName)
+			continue
+		}
+		if network.ManagementIP == "" {
+			continue
+		}
+
+		script := `function GetSourceVip {
+			param(
+					$NetworkName
+				)
+			$hnsNetwork = Get-HnsNetwork | ? Name -EQ $NetworkName.ToLower()
+    		$subnet = $hnsNetwork.Subnets[0].AddressPrefix
+			$ipamConfig = @"
+        {"cniVersion": "0.2.0", "name": "vxlan0", "ipam":{"type":"host-local","ranges":[[{"subnet":"$subnet"}]],"dataDir":"/var/lib/cni/networks"}}
+"@
+			$ipamConfig | Out-File "C:\k\sourceVipRequest.json"
```

Comment:
what is this `c:\k\` directory that's being used here? This isn't a path that we use anywhere else in the codebase.

Reply:
I think if I further fold it into pure Go, it might be able to go away. Let me try that.